lib/results.js

```diff
@@ -81,8 +81,14 @@ Results.prototype.createStream = function (opts) {
         self._stream.pipe(output);
     }
 
-    if (!this._isRunning) {
-        this._isRunning = true;
+    if (!self._isRunning) {
+        self._isRunning = true;
+        Results.__loadingPromise ? Results.__loadingPromise.then(run) : run();
```

Comment:
@ljharb is exposing the loading promise this way an acceptable way of waiting for all tests to be loaded before running them?

Reply:
It would be ideal not to expose it. Can this be a module-level variable instead?

Reply:
:thinking: not really sure what you mean by module-level variable here... the variable is set in `bin/tape.js`, but needs to accessed in `lib/results.js`

Reply:
Accessed how? Just reacting to the promise?

Reply:
yes

Reply:
So `bin/tape` creates the Promise, and `lib/results` needs to react to it?

Is there perhaps a way that either bin/tape could inject an executor into lib/results, or something?

Reply:
yes, precisely

well, `bin/tape` first needs to tell `lib/results` to wait and not immediately start the tests, and then it needs to say when they should be started.

perhaps we could implement a static function on Result to inject a promise instead of setting it manually?

```js
var loadingPromise = null;

Result.setLoadingPromise = function (promise) {
  loadingPromise = promise;
}

//

loadingPromise ? loadingPromise.then(run) : run();
```

that way would be more explicit.

Reply:
There might be a simpler way. What code calls Results.createStream in the first place? is there some way bin/tape could start the tests itself, and "pause" them, rather than needing static state and mutation?

Reply:
`createStream` is called in `createExitHarness`, https://github.com/substack/tape/blob/master/index.js#L62.

i tested adding a `wait` option to `getHarness`, and calling `tape.getHarness({ wait: true })` at the start of `bin/tape`. I also exposed a `tape.run()` method that creates the stream which i call when the loadingPromise has been resolved in `bin/tape`.

One negative side effect of this that would need to be fixed is that custom streams could not be created overriding the default stream as it will already have been created.

Another option would be to implement `tape.wait()` which set an a boolean and gets passed to `getHarness` internally.

Reply:
all those changes are in https://github.com/substack/tape/pull/547/commits/4f9787a71f259b179bef84ad4829dc51b5da380b

Reply:
since a bit of logic would have to be rethought to enable overriding the default stream after calling `getHarness`, i changed the logic to using a `wait` method and an internal wait boolean.